src/utilities/ArrayUtilities.ts

```diff
@@ -156,6 +156,10 @@ export namespace ArrayUtilities {
      * @return {T[][]} The array of subsets of array.
      */
     export function breakArrayIntoSubsets<T>(arr: T[], size: number): T[][] {
+        if(!arr){
```

Comment:
Why do we need this? The parameter `T[]` should guarantee some array, not `null` or `undefined` or anything like that.

Reply:
Yeah it was a bad attempt to fix it crashing on an empty arr, I worked around it later but forgot to remove this.  Will clean up.

Reply:
that being said, I recommend revisiting this function so it doesnt lead to crashes on certain input.

Reply:
Under what condition would this function lead to an error?